frontend/packages/knative-plugin/src/utils/fetch-dynamic-eventsources-utils.ts

```diff
@@ -26,58 +24,68 @@ let eventSourceModels: K8sKind[] = defaultEventSourceModels;
   const sortModels = _.orderBy(allModels, ['kind'], ['asc']);
-  const knownSourcesCrd = _.filter(
-    sortModels,
-    (model) =>
-      !!_.find(defaultEventSourceModels, { kind: model?.kind }) &&
-      model?.kind !== EventSourceCamelModel.kind,
-  );
+  const knownSourcesList = [
+    EventSourceApiServerModel.kind,
+    EventSourceContainerModel.kind,
+    EventSourceCronJobModel.kind,
+    EventSourceKafkaModel.kind,
+    EventSourcePingModel.kind,
+    EventSourceSinkBindingModel.kind,
+  ];
+  const knownSourcesCrd = _.filter(sortModels, (model) => knownSourcesList.includes(model?.kind));
   const camelSourcesCrd = _.filter(
     sortModels,
     (model) => model?.kind === EventSourceCamelModel.kind,
   );
   const dynamicSourcesCrd = _.filter(
     sortModels,
     (model) =>
-      !_.find(defaultEventSourceModels, { kind: model?.kind }) &&
-      model?.kind !== EventSourceCamelModel.kind,
+      !knownSourcesList.includes(model?.kind) && model?.kind !== EventSourceCamelModel.kind,
   );
   return [...knownSourcesCrd, ...camelSourcesCrd, ...dynamicSourcesCrd];
 };
 
 export const fetchEventSourcesCrd = async () => {
   let eventSourceModelList: K8sKind[] = [];
-  const url = `api/kubernetes/apis/${CustomResourceDefinitionModel.apiGroup}/${
-    CustomResourceDefinitionModel.apiVersion
-  }/${CustomResourceDefinitionModel.plural}?limit=250&labelSelector=${encodeURIComponent(
-    'duck.knative.dev/source=true',
-  )}`;
+  const url = 'api/console/knative-event-sources';
   try {
     const res = await coFetch(url);
     const resolvedRes = await res.json();
-    const allModels = _.map(resolvedRes?.items, (crd) => {
-      const {
-        spec: {
-          group,
-          version,
-          names: { kind, plural, singular },
-        },
-      } = crd;
-      return {
-        apiGroup: group,
-        apiVersion: version,
-        kind,
-        plural,
-        id: singular,
-        label: singular,
-        labelPlural: plural,
-        abbr: kindToAbbr(kind),
-        namespaced: true,
-        crd: true,
-        color: knativeEventingColor.value,
-      };
-    });
-    eventSourceModelList = orderedEventSourceModelData(allModels);
+    const allModels = _.reduce(
+      resolvedRes?.items,
+      (accumulator, crd) => {
+        const {
+          spec: {
+            group,
+            versions,
+            names: { kind, plural, singular },
+          },
+        } = crd;
+        const { name: version } = versions?.find((ver) => ver.served && ver.storage);
+        let modelData = {};
+        if (version) {
+          modelData = {
+            apiGroup: group,
+            apiVersion: version,
+            kind,
+            plural,
+            id: singular,
+            label: singular,
+            labelPlural: plural,
+            abbr: kindToAbbr(kind),
+            namespaced: true,
+            crd: true,
+            color: knativeEventingColor.value,
+          };
+        }
```

Comment:
```suggestion
        const modelData = version ? {
            apiGroup: group,
            apiVersion: version,
            kind,
            plural,
            id: singular,
            label: singular,
            labelPlural: plural,
            abbr: kindToAbbr(kind),
            namespaced: true,
            crd: true,
            color: knativeEventingColor.value,
          } : {};
```

Reply:
Ah, I see Sahil had exact opposite of my suggestion. Maybe a little different. :P